client/src/states/modules/map.js

```diff
@@ -9,18 +10,42 @@ const state = {
 }
 
 const getters = {
-
+  hasMap (state) {
+    return !! state.maps[0]
+  }
 }
 
 const mutations = {
-
+  updateMaps (state, { params }) {
+    state.maps = params.maps
+//    console.log(state.maps)
+  }, 
+  updateCurrentMap (state, newAddr) {
+    let today = new Date()
+    let time = "New Map on "+(today.getMonth()+1)+'/'+today.getDate()+'/'+today.getFullYear()+' '+today.getHours()+':'+today.getMinutes()
+    
+    state.currentMap = new Map(time, newAddr)   
+    state.maps.push(state.currentMap)
+//    console.log(state.currentMap)
+//    console.log(state.maps)
+  },
+  navigateCurrentMap (state, reqAddr) {
```

Comment:
We can change it to use the card store the index of map object and find it using index in dom.